src/NJsonSchema.CodeGeneration.TypeScript/TypeScriptTypeResolver.cs

```diff
@@ -149,9 +149,19 @@ private string Resolve(JsonSchema schema, string typeNameHint, bool addInterface
                 if (resolvedType != defaultType)
                 {
                     var keyType = Settings.TypeScriptVersion >= 2.1m ? prefix + resolvedType : defaultType;
-                    if (keyType != defaultType)
+                    if (keyType != defaultType && schema.DictionaryKey.ActualTypeSchema.IsEnumeration)
```

Comment:
This looks like a breaking change, no?

Reply:
I do not think so. I tried to find another type where the construct `key in keyof typeof <Identifier>` is useful.
With classes and interfaces it is actually only `key in keyof <Identifier>`. For union types it is only `key in <Identifier>`.

Therefore I do not think it's BC break. I see it more like better safe than sorry.